mountpoint-s3/scripts/fs_bench.sh

```diff
@@ -117,22 +224,38 @@ read_benchmark () {
     # run the benchmark
     run_fio_job $job_file $bench_file $mount_dir $log_dir
 
-    # unmount file system
-    sudo umount ${mount_dir}
+    cleanup
 
-    # cleanup mount directory and log directory
-    rm -rf ${mount_dir}
-    rm -rf ${log_dir}
   done
 }
 
 write_benchmark () {
   jobs_dir=mountpoint-s3/scripts/fio/write
 
   for job_file in "${jobs_dir}"/*.fio; do
+
+    if ! should_run_job "${job_file}"; then
+      echo "Skipping job ${job_file} because it does not match ${JOB_NAME_FILTER}"
+      continue
+    fi
+
     job_name=$(basename "${job_file}")
     job_name="${job_name%.*}"
     log_dir=logs/${job_name}
+
+
+    # cleanup mount directory and log directory
+    cleanup() {
+      echo "write_benchmark:cleanup"
+      # unmount file system only if it is mounted
+      ! mountpoint -q ${mount_dir} || sudo umount ${mount_dir}
```

Comment:
For reference[1]. In the current setup and if all job succeeds, cleanup will be called twice for the last job that runs. That is because it is called explicitly on line 283 (in this file) and by the trap. When that happens, the second `umount` will fail. To avoid redirecting all outputs of `umount` because that could mask other issues, I added this check to first check if it is mounted before attempting an un-mount.

I added the trap because when running locally, multiple times I ran into failure situations where the job failed and the cleanup never happened. So, I thought it was worth it to add this extra _complication_.

[1] https://man7.org/linux/man-pages/man1/mountpoint.1.html